packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -42,48 +42,44 @@ class Search extends React.Component {
       });
     } else {
       console.warn('Search has failed to load and now is being disabled');
-      this.setState({enabled: false});
+      setIsEnabled(false);
     }
-  }
-
-  toggleSearchIconClick() {
-    this.setState(
-      oldState => ({
-        isExpanded: !oldState.isExpanded,
-      }),
-      () => {
-        this.searchBarRef.current.focus();
-        this.props.handleSearchBarToggle();
-      },
-    );
-  }
+  }, []);
 
-  render() {
-    const {enabled, isExpanded} = this.state;
+  const toggleSearchIconClick = useCallback(() => {
```

Comment:
@yangshun Is it a good idea to wrap this function with `useCallback` ? The only case when this will re-render is when `props.isSearchBarExpanded` changes. Probabilty of re-rendering because of change of `isEnabled` is very low.

Reply:
It's fine to leave it as-is